File: 22-06-2024_Extract the Number from the String.cpp
class Solution {
  public:
    long long ExtractNumber(string sentence) {
        regex regex("\\d+");
        smatch match;
        string s = sentence;
        vector<string> numbers;

        while (regex_search(s, match, regex)) {
            numbers.push_back(match.str());
            s = match.suffix().str();
        }

        long long ans = -1;
        long long lens = LLONG_MIN;

        for (const auto& num : numbers) {
            if (num.find('9') == string::npos) {
                long long numValue = stoll(num);
                if (lens < numValue) {
                    ans = numValue;
                    lens = numValue;
                }
            }
        }

        return ans;
    }
};

File: 22-06-2024_Extract the Number from the String.java
class Solution {
    long ExtractNumber(String sentence) {
        // Define regex to find numbers
        Pattern pattern = Pattern.compile("\\d+");
        Matcher matcher = pattern.matcher(sentence);

        List<String> numbers = new ArrayList<>();

        // Extract all numbers from the string
        while (matcher.find()) {
            numbers.add(matcher.group());
        }

        long ans = -1;
        long maxVal = Long.MIN_VALUE;

        // Find the largest number that does not contain '9'
        for (String num : numbers) {
            if (!num.contains("9")) {
                long numValue = Long.parseLong(num);
                if (maxVal < numValue) {
                    ans = numValue;
                    maxVal = numValue;
                }
            }
        }

        return ans;
    }
}
